scripts/test/Muon/utilities/muon_workspace_test.py

```diff
@@ -0,0 +1,387 @@
+        six.assertCountEqual(self, ws_property.readY(0), [10, 10, 10, 10])
+
+    def test_that_workspace_property_returns_workspace_when_in_ADS(self):
+        workspace_handle = MuonWorkspace(workspace=self.workspace)
+
+        workspace_handle.show("arbitrary_name")
+        ws_property = workspace_handle.workspace
+
+        six.assertCountEqual(self, ws_property.readX(0), [1, 2, 3, 4])
+        six.assertCountEqual(self, ws_property.readY(0), [10, 10, 10, 10])
+
+    def test_that_can_change_name_when_workspace_not_in_ADS(self):
+        workspace_handle = MuonWorkspace(workspace=self.workspace)
+
+        workspace_handle.name = "new_name"
+
+        self.assertEqual(workspace_handle.name, "new_name")
+
+    def test_that_running_show_twice_with_different_names_causes_the_workspace_to_be_moved(self):
+        workspace_handle = MuonWorkspace(workspace=self.workspace)
+
+        workspace_handle.show("name1")
+        workspace_handle.show("name2")
+
+        self.assertFalse(simpleapi.mtd.doesExist("name1"))
+        self.assertTrue(simpleapi.mtd.doesExist("name2"))
+
+    def test_that_cannot_change_name_when_workspace_in_ADS(self):
+        workspace_handle = MuonWorkspace(workspace=self.workspace)
+
+        workspace_handle.show("name1")
+
+        with self.assertRaises(ValueError):
+            workspace_handle.name = "new_name"
+
+    def test_that_hiding_workspace_more_than_once_has_no_effect_but_raises_RuntimeWarning(self):
+        workspace_handle = MuonWorkspace(workspace=self.workspace)
+        workspace_handle.show("name1")
+
+        workspace_handle.hide()
+
+        with self.assertRaises(RuntimeWarning):
+            workspace_handle.hide()
+
+    def test_that_if_workspace_deleted_from_ADS_then_hide_raises_a_RuntimeWarning(self):
+        workspace_handle = MuonWorkspace(workspace=self.workspace)
+        workspace_handle.show("name1")
+
+        simpleapi.mtd.clear()
+
+        with self.assertRaises(RuntimeWarning):
+            workspace_handle.hide()
+
+    def test_that_hiding_workspace_deletes_groups_which_are_left_empty(self):
+        # TODO
+        pass
+
+    def test_that_hiding_workspace_does_not_delete_groups_which_still_contain_workspaces(self):
+        # TODO
+        pass
+
+    # ----------------------------------------------------------------------------------------------
+    # Overwriting the workspace via the workspace property
+    # ----------------------------------------------------------------------------------------------
+
+    def test_that_setting_a_new_workspace_removes_the_previous_one_from_the_ADS(self):
+        workspace_handle = MuonWorkspace(workspace=self.workspace)
+        workspace_handle.show("name1")
+
+        workspace2 = create_simple_workspace(data_x=[5, 6, 7, 8], data_y=[20, 20, 20, 20])
+
+        self.assertTrue(simpleapi.mtd.doesExist("name1"))
+        workspace_handle.workspace = workspace2
+        self.assertFalse(simpleapi.mtd.doesExist("name1"))
```

Comment:
I would have expected it to just replace the workspace (so kept it in the ADS and replaced the data)